clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -81,6 +81,14 @@
     public List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
```

Comment:
I copied that documentation from `Selector.disconnect`. There is the following code in `Selector.poll`:

```
                    if (!key.isValid()) {
                        close(channel);
                        this.disconnected.add(channel.id());
                    }
```

And there is the following test in `NetworkClientTest`:

```
        assertTrue("Now the client is ready", client.ready(node, time.milliseconds()));
        selector.disconnect(node.idString());
        client.poll(1, time.milliseconds());
        selector.clear();
        assertFalse("After we forced the disconnection the client is no longer ready.", client.ready(node, time.milliseconds()));
        assertTrue("Metadata should get updated.", metadata.timeToNextUpdate(time.milliseconds()) == 0);
```

So, that lead me to believe that a poll was needed to update the internal state of the `Selector` and `NetworkClient`. The test above fails without the `poll` line (although it does use a `MockSelector`, so the bug could be there).

What are your thoughts based on this additional information?